core/shared/src/main/scala/zio/logging/internal/WriterProvider.scala

```diff
@@ -0,0 +1,75 @@
+package zio.logging.internal
+
+import java.io.{ BufferedWriter, FileOutputStream, OutputStreamWriter, Writer }
+import java.nio.charset.Charset
+import java.nio.file.{ FileSystems, Path }
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+
+private[logging] sealed trait WriterProvider {
+  def writer: Writer
+}
+
+private[logging] object WriterProvider {
+  final case class SimpleWriterProvider(
+    destination: Path,
+    charset: Charset,
+    bufferedIOSize: Option[Int]
+  ) extends WriterProvider {
+    override val writer: Writer = {
+      val output = new OutputStreamWriter(new FileOutputStream(destination.toFile, true), charset)
+      bufferedIOSize match {
+        case Some(bufferSize) => new BufferedWriter(output, bufferSize)
+        case None             => output
+      }
+    }
+  }
+
+  final case class TimeBasedRollingWriterProvider(
+    destination: Path,
+    charset: Charset,
+    bufferedIOSize: Option[Int]
+  ) extends WriterProvider {
+    import TimeBasedRollingWriterProvider._
+
+    private var timeInUse             = makeNewTime
+    private var currentWriter: Writer = makeWriter(makePath(destination, timeInUse), charset, bufferedIOSize)
+    val formatter: DateTimeFormatter  = DateTimeFormatter.ofPattern("yyyy-MM-dd")
```

Comment:
is this used ?

Reply:
Not use, I removed formatter in provider class